core/sdk-core/src/main/java/software/amazon/awssdk/core/sync/ResponseTransformer.java

```diff
@@ -78,7 +79,17 @@
      * @throws Exception if any error occurs during processing of the response. This will be re-thrown by the SDK, possibly
      *                   wrapped in an {@link SdkClientException}.
      */
-    ReturnT transform(ResponseT response, AbortableInputStream inputStream) throws Exception;
+    default ReturnT transform(ResponseT response, AbortableInputStream inputStream) throws Exception {
+        try {
+            return apply(response, inputStream);
+        } catch (RetryableException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new NonRetryableException(e);
+        }
+    }
+
+    ReturnT apply(ResponseT response, AbortableInputStream inputStream) throws Exception;
```

Comment:
hmm, not sure why checkstyle doesn't fail on this, but `apply` is conflicting with kotlin function name. Can we make `apply` default method and `transform` abstract method?

https://github.com/aws/aws-sdk-java-v2/blob/0e14d03bfab5440ef0380bf637fed9304e66e2c2/build-tools/src/main/resources/software/amazon/awssdk/checkstyle.xml#L255-L260